Gems/LyShine/Code/Editor/EditorWindow.cpp

```diff
@@ -697,15 +698,27 @@ bool EditorWindow::SaveCanvasToXml(UiCanvasMetadata& canvasMetadata, bool forceA
         else if (recentFiles.size() > 0)
         {
             dir = Path::GetPath(recentFiles.front());
-            dir.append(canvasMetadata.m_canvasDisplayName.c_str());
         }
         // Else go to the default canvas directory
         else
         {
             dir = FileHelpers::GetAbsoluteDir(UICANVASEDITOR_CANVAS_DIRECTORY);
-            dir.append(canvasMetadata.m_canvasDisplayName.c_str());
         }
 
+        // Make sure the directory exists. If not, walk up the directory path until we find one that does
+        // so that we will have a consistent 'starting folder' in the 'AzQtComponents::FileDialog::GetSaveFileName' call
+        // across different platforms. 
+        AZ::IO::BasicPath<AZStd::string> dirPath(dir.toUtf8().constData());
+        QDir validDirCheck(dirPath.c_str());
+        while (!validDirCheck.exists() && (dirPath.HasParentPath()))
```

Comment:
This needs to protect against the parent path being itself. 
Because a the parent of the path `/` is `/`

Reply:
I thought that was what 'HasParentPath' will do. But if not, then will add another check

Reply:
HasParentPath is indicating if the path actually has a parent.
The parent of the root path is actually the root path.

HasParentPath will return false for a relative path such as "foo", but for an absolute path such as `C:\foo`, the parent first is `C:\`, the next parent is **still `C:\`**